datalake/src/main/java/com/sequenceiq/datalake/flow/upgrade/ccm/UpgradeCcmActions.java

```diff
@@ -0,0 +1,127 @@
+package com.sequenceiq.datalake.flow.upgrade.ccm;
+
+import static com.sequenceiq.datalake.flow.upgrade.ccm.UpgradeCcmStateSelectors.UPGRADE_CCM_FAILED_HANDLED_EVENT;
+import static com.sequenceiq.datalake.flow.upgrade.ccm.UpgradeCcmStateSelectors.UPGRADE_CCM_FINALIZED_EVENT;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmFailedEvent;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmStackEvent;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmStackRequest;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmSuccessEvent;
+import com.sequenceiq.datalake.metric.MetricType;
+import com.sequenceiq.datalake.metric.SdxMetricService;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.FlowState;
+
+@Configuration
+public class UpgradeCcmActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpgradeCcmActions.class);
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxMetricService metricService;
+
+    @Bean(name = "UPGRADE_CCM_UPGRADE_STACK_STATE")
+    public Action<?, ?> upgradeCcmStack() {
+        return new AbstractSdxAction<>(UpgradeCcmStackEvent.class) {
+
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    UpgradeCcmStackEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, UpgradeCcmStackEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Execute CCM upgrade stack flow for SDX: {}", payload.getResourceId());
+                sendEvent(context);
```

Comment:
How about calling `sdxStatusService.setStatusForDatalakeAndNotify()` here? Or will it be done later in the event handler?